src/main/python/pre_fx_data.py

```diff
@@ -0,0 +1,269 @@
+        run the pre-fx pipeline
+
+        Parameters
+        ----------
+        path : 
+            load ontology from here
+
+        """
+
+        try:
+            with open(path, "r") as ontology_file:
+                ontology_data = json.load(ontology_file)
+            ontology = StimulusOntology(ontology_data)
+            
+            if self.nwb_path is not None and self.qc_criteria is not None:
+                self.run_pipeline(
+                    self.nwb_path, 
+                    ontology, 
+                    self.qc_criteria, 
+                    commit=True
+                )
+            else:
+                self.stimulus_ontology = ontology
+
+        except Exception as err:
+            exception_message(
+                "StimulusOntology load failed",
+                f"failed to load stimulus ontology file from {path}",
+                err
+            )
+
+    def load_qc_criteria_from_json(self, path: str):
+        """ Attempts to read qc criteria from a JSON. If successful (and other 
+        required data are already set), attempts to run the pre-fx pipeline
+
+        Parameters
+        ----------
+        path : 
+            load criteria from here
+
+        """
+
+        try:
+            with open(path, "r") as criteria_file:
+                criteria = json.load(criteria_file)
+            
+            if self.nwb_path is not None and self.stimulus_ontology is not None:
+                self.run_pipeline(
+                    self.nwb_path, 
+                    self.stimulus_ontology, 
+                    criteria, 
+                    commit=True
+                )
+            else:
+                self.qc_criteria = criteria
+
+        except Exception as err:
+            exception_message(
+                "QC criteria load failure",
+                f"failed to load qc criteria file from {path}",
+                err
+            )
+
+    def load_data_set_from_nwb(self, path: str):
+        """ Attempts to read an NWB file describing an experiment. Fails if 
+        qc criteria or stimulus ontology not already present. Otherwise, 
+        attempts to run the pre-fx pipeline.
+
+        Parameters
+        ----------
+        path : 
+            load dataset from here
+
+        """
+
+        try:
+            if self.stimulus_ontology is None:
+                raise ValueError("must set stimulus ontology before loading a data set!")
+            elif self.qc_criteria is None:
+                raise ValueError("must set qc criteria before loading a data set!")
+            
+            self.run_pipeline(path, self.stimulus_ontology, self.qc_criteria, commit=True)
+
+        except Exception as err:
+            exception_message(
+                "Unable to load NWB",
+                f"failed to load NWB file from {path}",
+                err
+            )
+
+
+    def run_pipeline(self, nwb_path, stimulus_ontology, qc_criteria, commit=True):
```

Comment:
I believe `run_pipeline` name is confusing. Here we are running QC, thus it would be better to call this run_qc. The pipeline also includes fx stage and manual qc. So this is confusing.

Reply:
`run_extraction_and_qc`?

Reply:
mouthful: `run_extraction_and_auto_qc`